pkg/controllers/dynakube/version/oneagent.go

```diff
@@ -101,3 +102,19 @@ func (updater *oneAgentUpdater) CheckForDowngrade(latestVersion string) (bool, e
 	}
 	return isDowngrade(updater.Name(), previousVersion, latestVersion)
 }
+
+func (updater oneAgentUpdater) ValidateStatus() error {
+	imageVersion := updater.Target().Version
+	imageType := updater.Target().Type
+
+	if imageVersion == "" {
+		return errors.New("build version of OneAgent image not set")
+	}
+
+	if imageType == status.ImmutableImageType && (updater.dynakube.FeatureDisableReadOnlyOneAgent() || updater.dynakube.ClassicFullStackMode()) {
+		return errors.New("immutable OneAgent image in combination with classicFullStack mode or readOnly OneAgent filesystem is not possible")
+	}
+	log.Info("successfully validated OneAgent image")
```

Comment:
```suggestion
	log.Info("OneAgent metadata present")
```
or some other way to inform the user on what was validated

Reply:
done in https://github.com/Dynatrace/dynatrace-operator/pull/2156/commits/27899211fb4112ad1045ccc88b8e491f28168227